site/community/release_guide.md

```diff
@@ -0,0 +1,475 @@
+To "dry run", you can provide "-DdryRun" at the end of this command. "dry run" will generate some temporary files in the project folder, you can remove them by running "mvn release:clean".
+
+    mvn release:branch \
+        -DbranchName=${BRANCH_NAME} \
+        -DdevelopmentVersion=${DEVELOPMENT_VERSION} \
+        [-DdryRun]
+
+However, if you are doing an incremental/hotfix release, please run the following command after checking out the release tag of the release being patched.
+
+    mvn release:branch \
+        -DbranchName=${BRANCH_NAME} \
+        -DupdateWorkingCopyVersions=false \
+        -DupdateBranchVersions=true \
+        -DreleaseVersion="${VERSION}-SNAPSHOT" \
+        [-DdryRun]
+
+Check out the release branch.
+
+    git checkout ${BRANCH_NAME}
+
+The rest of this guide assumes that commands are run in the root of a repository on `${BRANCH_NAME}` with the above environment variables set.
+
+### Checklist to proceed to the next step
+
+1. Release Manager’s GPG key is published to `dist.apache.org`
+2. Release Manager’s GPG key is configured in `git` configuration
+3. Release Manager has `org.apache.bookkeeper` listed under `Staging Profiles` in Nexus
+4. Release Manager’s Nexus User Token is configured in `settings.xml`
+5. JIRA release item for the subsequent release has been created
+6. Github milestone item for the subsequet release has been created
+7. There are no release blocking JIRA issues
+8. There are no release blocking Github issues
+9. Release Notes in JIRA have been audited and adjusted
+10. Release Notes for Github Milestone is generated, audited and adjusted
+11. Release branch has been created
+12. Originating branch has the version information updated to the new version
+
+**********
+
+## Build a release candidate
+
+The core of the release process is the build-vote-fix cycle. Each cycle produces one release candidate. The Release Manager repeats this cycle until the community approves one release candidate, which is then finalized.
+
+### Build and stage Java artifacts with Maven
+
+
+    TODO: Currently we have to build and stage maven artifacts manually, because it requires pushing the artifacts to apache staging. We should look for a solution to automate that.
+
+
+Set up a few environment variables to simplify Maven commands that follow. This identifies the release candidate being built. Start with `release candidate number` equal to `0` and increment it for each candidate.
+
+    RC_NUM="0"
+    TAG="release-${VERSION}"
+    RC_DIR="bookkeeper-${VERSION}-rc${RC_NUM}"
+
+Use Maven release plugin to build the release artifacts, as follows:
+
+    mvn release:prepare \
+        -Dresume=false \
+        -DreleaseVersion=${VERSION} \
+        -Dtag=${TAG} \
+        -DupdateWorkingCopyVersions=false \
+        [-DdryRun] [-DskipTests]
+
+Use Maven release plugin to stage these artifacts on the Apache Nexus repository, as follows:
+
+    mvn release:perform [-DdryRun] [-DskipTests]
```

Comment:
on failures:

- delete the release tag: git tag -d ${VERSION} (locally) git push apache :refs/tags/${VERSION}
- and run 'release:prepare' again.